components/automate-ui/src/app/modules/infra-proxy/treetable/component/treetable.component.spec.ts

```diff
@@ -0,0 +1,42 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { TreetableComponent } from './treetable.component';
+import { MatTableModule } from '@angular/material/table';
+import { MatIconModule } from '@angular/material/icon';
+import { mockSearchableTree } from '../mocks/mockSearchableTree';
+import * as _ from 'lodash';
+
+describe('TreetableComponent', () => {
+  let component: TreetableComponent<any>;
+  let fixture: ComponentFixture<TreetableComponent<any>>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        TreetableComponent
+      ],
+      imports: [
+        MatTableModule,
+        MatIconModule
+      ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(TreetableComponent);
+    component = fixture.componentInstance;
+    component.tree = _.cloneDeep(mockSearchableTree);
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should emit an event when a node is clicked', () => {
+    const clickedNode = (component as any).treeTable[0];
+    component.nodeClicked.subscribe(n => expect(n).toBe(clickedNode));
+    component.onNodeClick(clickedNode);
+  });
+
```

Comment:
Please add tests. This is another complicated component that needs some. Thx!

Reply:
This file covered the cases for node clicked as per component code and all other test covered inside the helper component.